### PR TITLE
Add support for puppeteer-core

### PR DESCRIPTION
**Pull request by MickaelTH**
## Motivation/Description of the PR

I needed to use with Browserless so only puppeteer-core is required in this case, not need to install 100+Mb of Chromium bundled with puppeteer.

Applicable helpers:

- [ ] Webdriver
- [ X] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- Description of this PR, which problem it solves
- A link to the corresponding issue (if applicable).

## Type of change

- [ ] Breaking changes
- [ X] New functionality
- [x] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [ ] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [ ] Lint checking (Run `npm run lint`)
- [ ] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/helper/Puppeteer.js [hunk 1]
```diff
@@ -148,8 +147,8 @@ const consoleLogStore = new Console();
 class Puppeteer extends Helper {
   constructor(config) {
     super(config);
-
-    puppeteer = requireg(config.browser === 'firefox' ? 'puppeteer-firefox' : 'puppeteer');
+
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Codacy Issue found: [Trailing spaces not allowed. (no-trailing-spaces)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4646938)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4646938&bid=15562235)

File: lib/helper/Puppeteer.js [hunk 2]
```diff
@@ -148,8 +147,8 @@ const consoleLogStore = new Console();
 class Puppeteer extends Helper {
   constructor(config) {
     super(config);
-
-    puppeteer = requireg(config.browser === 'firefox' ? 'puppeteer-firefox' : 'puppeteer');
+    
+    puppeteer = requireg(config.browser === 'firefox' ? 'puppeteer-firefox' : (config.chrome.browserWSEndpoint ? 'puppeteer-core' : 'puppeteer'));
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js]: Codacy Issue found: [Do not nest ternary expressions. (no-nested-ternary)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4646938)

File: lib/helper/Puppeteer.js [hunk 3]
```diff
@@ -13,7 +13,6 @@ const {
   ucfirst,
   fileExists,
   chunkArray,
-  toCamelCase,
```
Comment (PeterNgTr) [inline lib/helper/Puppeteer.js]: I thought this was in this PR https://github.com/Codeception/CodeceptJS/pull/2059/files#diff-0fff3879242c78dedc81559a6091f25bR16. Any reason for removing this?

Comment (MickaelTH) [reply lib/helper/Puppeteer.js]: That's interesting, I did not remove or touch this line. I will try to redo a better PR.

Comment (DavertMik) [reply lib/helper/Puppeteer.js]: Yes, this was a part of a previous fix.

**Comment by MickaelTH** [comment]
As informed on the issue #2076, it is easier to add param in npm config to skip the installation of Chromium. Therefore no change required.

**Pull request closed without merging**